js/widgets/tempo.js

```diff
@@ -255,6 +255,13 @@ class Tempo {
      * @returns {void}
      */
     _useBPM(i) {
+        const input = this.BPMInputs[i].value;
+
+        if (isNaN(input)) {
+            this.activity.errorMsg(_("Invalid input. Please enter a valid number."));
```

Comment:
Maybe it would make more sense to say something along the lines of "Please enter a number between 30 and 1000" ???

Reply:
sure .